can/CMakeLists.txt

```diff
@@ -2,12 +2,36 @@
 # Authors: Daniele E. Domenichelli <daniele.domenichelli@iit.it>
 # CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
 
-
-project(canProtocolLib NONE)
 if(WITH_CANPROTOCOLLIB)
+  set(LIBRARY_TARGET_NAME canProtocolLib)
+  
+  set(${LIBRARY_TARGET_NAME}_HDR ${CMAKE_CURRENT_SOURCE_DIR}/canProtocolLib/iCubCanProto_analogSensorMessages.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/canProtocolLib/iCubCanProto_bootloaderMessages.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/canProtocolLib/iCubCanProto_classes.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/canProtocolLib/iCubCanProtocol.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/canProtocolLib/iCubCanProto_inertialSensorMessages.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/canProtocolLib/iCubCanProto_motorControlMessages.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/canProtocolLib/iCubCanProto_skinMessages.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/canProtocolLib/iCubCanProto_types.h)
+  
+  add_library(${LIBRARY_TARGET_NAME} INTERFACE)
+  
+  add_library(${PROJECT_NAME}::${LIBRARY_TARGET_NAME} ALIAS ${LIBRARY_TARGET_NAME})
+  
+  target_include_directories(${LIBRARY_TARGET_NAME} INTERFACE "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/${LIBRARY_TARGET_NAME}>"
+                                                              "$<INSTALL_INTERFACE:${icub_firmware_shared_INSTALL_INCLUDE_DIR}/${LIBRARY_TARGET_NAME}>")
+
+  if(${CMAKE_VERSION} VERSION_LESS "3.15.0")
+    install(TARGETS ${LIBRARY_TARGET_NAME}
+            EXPORT  ${PROJECT_NAME})
```

Comment:
This is fishy. The `install(TARGETS` with default destination is only available since 3.14, see https://cmake.org/cmake/help/v3.14/command/install.html#installing-targets . Before of that, you  had to explicitly set the destinations, see https://github.com/robotology/how-to-export-cpp-library/blob/master/src/LibTemplateCMake/CMakeLists.txt#L88 . I am afraid that in this stage the project would not compile with CMake 3.12 and 3.13 , so: 
* We need to decide if we want to support CMake 3.12 and 3.13 
* We need to update the `cmake_minimum_required` version once we decided which version to support

Reply:
In `icub-main` we have  `3.12` as `cmake_minimum_required`, I'd keep the same also here

Reply:
The only problem is, on the `icub-head` we have should have this version of `cmake` right?
I am worried about the old one as `iCubGenova01
`

Reply:
I think we have already sorted that out when  we required CMake 3.12 for YARP.

Reply:
This https://github.com/robotology/icub-firmware-shared/pull/35/commits/05662093533a40f384f861a357a204a73e81cad5 add the `cmake_minimum_required` to `3.12`

I tested on  a docker and with cmake 3.12.4 it compiles and install fine

Reply:
Ok, that is fishy as it does not seems  to be coherent with the docs, but if the problem is only on Windows, it is not a big problem as in any case Windows user have recent CMake  versions.